### DISCUSSION TITLE
Unable to resolve module node:http

### DISCUSSION THREAD

**Discussion by kabundege** [node-fetch/node-fetch#1504] [Q&A]
# Description?
I'm working on a ReactNative app, and I was trying to make an HTTP request on an API.
The implementation in use works fine (I've been using it for a while now).

![Screen Shot 2022-01-29 at 20 46 47](https://user-images.githubusercontent.com/50319826/151673573-d6dbb1bf-2950-4dd5-aa0a-1cd10caccde4.png)

But with this new version `3.2.0`, when i import the file, I immediately get
`Unable to resolve module node:http` in my the console `cmd`,
and also this suggestion below.

**If you are sure the module exists, try these steps:**
 1. Clear watchman watches: watchman watch-del-all
 2. Delete node_modules and run yarn install
 3. Reset Metro's cache: yarn start --reset-cache
 4. Remove the cache: rm -rf /tmp/metro-*

**Reproduction**

1.  yarn add node-fetch
2. run the code above ( image )

**Expected behavior**

**Screenshots**
![Screen Shot 2022-01-29 at 20 51 57](https://user-images.githubusercontent.com/50319826/151673787-1769346c-8a1b-42d7-b8ef-10f0fb7e8e72.png)

**Your Environment**

| software         | version
| ---------------- | -------
| node-fetch       | 3.2.0
| node             | 16.13.1
| npm              | 8.1.2
| Operating System | Mac OS X (macOS)

**Additional context**

**Comment by jimmywarting** [comment]
This is a problem with React Native and the bug should be reported to facebook or whatever package they try to use to resolve npm/node packages.

**Reply by jimmywarting** [reply]
converted this into a discussion cuz there are currently no plan to fix/change any of this... ppl have asked us to add node: prefix already, and we can't support both worlds by omiting the node: prefix now. cuz that breaks other build tools that rely on a old static list of built in node modules

by supplying a `node:` prefix then some npm packages don't need to keep this list updated, they could just look at the prefix to figure this out if it is a npm package or a built in node module... this was the case when we depended on a new node dependency that the package haven't yet added to the list. cuz it was stalled (not updated)

**Comment by kabundege** [comment]
> This is a problem with React Native and the bug should be reported to Facebook

## Work-around?
if in need, reverted back to node-fetch `^2.6.1` , it works fine

**Comment by PowVT** [comment]
reverting back to 2.6.1 worked for me, I had a similar error where 'node:buffer could not be found'

**Comment by LinusU** [comment]
React Native have `fetch` builtin already, you shouldn't need this package at all?

This package is for Node.js and won't work in other environments since it uses builtin packages in Node.js to issue http/https requests...

**Comment by kabundege** [comment]
> React Native have `fetch` builtin already, you shouldn't need this package at all?
> 
> This package is for Node.js and won't work in other environments since it uses built-in packages in Node.js to issue http/https requests...

I get what you're saying,
but maybe you could look into it, because the error doesn't occur on node-fetch `2.6.1` or lower

**Comment by LinusU** [comment]
@kabundege what happens if you just remove `import fetch from 'node-fetch'`? And just use `fetch` directly, which will be the React Native version.

**Reply by Anatame** [reply]
> @kabundege what happens if you just remove `import fetch from 'node-fetch'`? And just use `fetch` directly, which will be the React Native version.

Hey I was facing this error and as soon as I removed the import and tried using fetch directly, it worked! How do I make this normal?

**Reply by coderdev27** [reply]
React native has built in fetch, uninstall node-fetch package. Command for uninstalling - npm uninstall node-fetch

**Comment by torressam333** [comment]
Yeah this is an issue with the latest node-fetch (3.3.1) as of time of this comment. I am back on 2.6.1 and the error goes away. I got this issue when trying to use it in a Nuxt 2 app.

**Reply by jimmywarting** [reply]
Using `node:` prefix isn't really an issue in itself. it's ok to use it.
There where a reason why we switched to using `node:` prefix in the first place. as we depended on an optional `node:stream/web` from NodeJS but where not using `node:` prefix at that time. He argued that it wasn't able to resolve the module `"stream/web"` cuz it couldn't find a package with that name. if only we hade used `node:` prefix then it would had know that it came from NodeJS core itself and not from some npm package.

I would probably say that there is something wrong with how Nuxt or and sub (dev)dependencies tries to resolve packages.

**Reply by jimmywarting** [reply]
could you create a minimal reproduction of the error? maybe from using [stackblitz](https://stackblitz.com/edit/node-s6cpgd?file=index.js)
